rank/pages/api/search.ts

```diff
@@ -1,35 +1,32 @@
+import { Namespace, namespaces } from '../../types/namespace'
 import { NextApiRequest, NextApiResponse } from 'next'
-import { SearchResponse, rankClient } from '../../services/elasticsearch'
-import { TestResult, runTests } from './eval'
+
+import { SearchResponse } from '../../types/elasticsearch'
+import { TestResult } from '../../types/test'
+import { getRankClient } from '../../services/elasticsearch'
 import ranks from '../../ranks'
-import { Namespace } from '../../types/namespace'
+import { runTests } from './eval'
 
 export type ApiResponse = SearchResponse & {
   results: TestResult[]
 }
 
-export type ApiRequest = {
-  query?: string
-  useTestQuery?: 'true' | 'false'
-  namespace?: Namespace
-}
-
-type Q = NextApiRequest['query']
-const decoder = (q: Q) => ({
-  query: q.query ? q.query.toString() : undefined,
-  rankId: q.rankId ? q.rankId.toString() : 'works-prod',
+const decoder = (req: NextApiRequest) => ({
+  query: req.query.query ? req.query.query.toString() : undefined,
+  namespace: req.query.namespace ? req.query.namespace : 'works',
```

Comment:
we have `decodeNamespace` and could create a `decodeEnv`
https://github.com/wellcomecollection/catalogue-api/blob/6272d56aa00c222f9a76541ad7e8bd4e48fb2fc5/rank/types/decoder.ts#L13-L16

Reply:
👍

Reply:
something like this?
```ts
const decoder = (req: NextApiRequest) => ({
  query: req.query.query ? decodeString(req.query, 'query') : undefined,
  namespace: req.query.namespace ? decodeNamespace(req.query.namespace) : 'works',
  env: req.query.env ? decodeEnv(req.query.env) : 'prod',
})
```

Reply:
Yea, I was working on a wrapper method, but couldn't quite get the typing correct - similar to:

```ts
{ query: optional(decodeString(req.query)) }
```

Reply:
ah nice - will keep this in mind